public/apps/login/test/__snapshots__/login-page.test.tsx.snap

```diff
@@ -121,7 +121,7 @@ exports[`Login page renders renders with config value for multiauth 1`] = `
         aria-label="openid_login_button"
         className="test-btn-style"
         data-test-subj="submit"
-        href="/app/opensearch-dashboards/auth/openid/login"
+        href="/app/opensearch-dashboards/auth/openid/captureUrlFragment?nextUrl=%2F"
```

Comment:
curious as to whether captureUrlFragment is just an arbitrary string that you are using here to determine something, or whether it has some other meaning?

Reply:
If a user directly tries to go to a page like `http://localhost:5601/app/security-dashboards-plugin#/getstarted`, but is not authenticated yet they are re-directed to the login screen. This PR intercepts that re-direct to capture the hash in the URL in localStorage before the user is re-directed out of OSD to the external IdP's login screen. When they successfully login they are re-directed back to OSD and the re-direct is intercepted by `/auth/openid/redirectUrlFragment` which gets the hash from local storage and routes the user back to that page. 

This implementation is borrowed from the SAML implementation for the same feature. That reminds me that the SAML implementation had to be updated to account for multi-auth so as a follow-up to this PR I will add tests that verify this works when multiauth is enabled.

Reply:
PR that updated SAML to work with multiauth enabled: https://github.com/opensearch-project/security-dashboards-plugin/pull/1557

Reply:
Let me look at that PR as well for full context

Reply:
FYI this PR does support multi-auth as well, but does not explicitly add tests with multi-auth. Similar to SAML, there should be tests that assert that this behavior works with multiauth enabled as well.

There's an open issue to convert the SAML multiauth tests to cypress: https://github.com/opensearch-project/security-dashboards-plugin/issues/1667